Fixes #1 and #4

Osterjour opened the pull request:
The setImmediate (#4) fixed also a huge problem with loosing some lines if you try to pause and resume after every line.

Osterjour committed:
Fixing #1 and #4

The setImmediate (#4) fixed also a huge problem with loosing some lines if you try to pause and resume after every line.

bblack commented:
:+1: This fixes dropping a bunch of lines at the end of a file when i'm pausing and resuming after each line.

mbochynski commented:
This fix help me also. :+1: Thanks so much. I spend much time inspecting my code...

Osterjour committed:
Returning to process.nextTick (we always used this due to a bug)
Added .idea to gitignore
Added an example

Osterjour committed:
Sorry guys! (I shouldn't work on this without being really focused)
- Reverting the process.nextTick to something which should really work
- Adding a function to work around multiple emits of 'end'

Osterjour committed:
Fix a bug in README examples

Osterjour commented:
@bladepan Thanks for pointing that out! After messing things really up I hopefully fixed this issue now.
Would be great if you could check with the newest version in my branch and give me feedback.

About pushing this to the npmjs module: I already wrote an email to @RustyMarvin about moving the npmjs module to my branch. If I don't get an answer from him in the next 4 weeks I will ask the npmjs guys to give me the needed permissions to switch over the npmjs module.

Osterjour committed:
I added one empty line to have the 5100 lines with skipEmptyLines=false

Osterjour committed:
Switching to own repo to maintain the package from now on

Osterjour committed:
We fixed this bug - so we can remove it from the known bugs list

Osterjour committed:
Adding travis-ci and jshint as first test

Osterjour committed:
bump version to 0.1.3

Osterjour committed:
Adding badges to README

Osterjour committed:
Adding npm update for travis to get the jshint running

alexloi committed:
Added open emit

netoneko committed:
Add support for start/end options for file stream

Osterjour committed:
Merge pull request #1 from StreetHub/streethub

Added open emit

Osterjour committed:
Merge pull request #2 from netoneko/master

Add support for start/end options for file stream

gogson committed:
[fix] end event was emitted when the stream read all the file, not when line-by-line reached the last line

Osterjour committed:
Merge pull request #6 from LeanJS/master

[fix] end event was emitted when the stream read all the file, not wh…

Osterjour committed:
push version number to 0.1.5

Osterjour committed:
adding new node versions for travis

Commit:
Cleared internal lines buffer so close could be achieved

In the 'close' method if the _lines array is not cleared, i.e. set to []
then further on in the _nextLine() method you never get to the
if(this._end) which if passed causes the actual stop of emitting the
'line' event.

Osterjour committed:
Merge pull request #7 from mysticx/master

Cleared internal lines buffer so close could be achieved

abelozerov committed:
Support for initialization with stream instead of file name. Support for buffered _readStream data

abelozerov committed:
Too early end emit when paused fixed

abelozerov committed:
Merge branch 'bug/incorrect_async_end'

Osterjour committed:
Merge pull request #13 from abelozerov/bug/incorrect_async_end

Too early end emit when paused fixed

Osterjour committed:
Merge pull request #12 from abelozerov/master

Support for initialization with stream along with file name. Support for buffered _readStream data

gbejic committed:
Update README.md

I was assuming that its start and end line number.

Osterjour committed:
Merge pull request #17 from gbejic/patch-1

Update README.md

Commit:
- Update travis to check only from 4.x on
- Push version number to update NPM with the patches
